File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> complementMap;
        vector<int> output;
        int complement = 0;
        for(int i = 0; i < nums.size(); i++){
            complement = target - nums[i];
            if(complementMap.find(complement) != complementMap.end()){
                output = {complementMap[complement],i};
                return output;
            }
            complementMap[nums[i]] = i;
        }
        return output;
        
    }
};

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        goal = len(nums)-1
        for r in range(len(nums) -2, -1, -1):
            if r + nums[r] >= goal: 
                goal = r
        return goal == 0
    

        